src/datasets/data_files.py

```diff
@@ -583,7 +583,8 @@ def from_patterns(
                     )
                 )
             except FileNotFoundError:
-                pass
+                if "*" not in pattern and "[" not in pattern and "?" not in pattern:
```

Comment:
The `glob` module `has_magic` (public, but not documented IIRC; e.g., `fsspec` uses it) to check this:
```suggestion
                if has_magic(pattern):
```

But feel free to decide which one you prefer.